observing/obsstate.py

```diff
@@ -0,0 +1,189 @@
+
+class Session(BaseModel):
+    PI_ID: str
+    PI_NAME: str
+    PROJECT_ID: str
+    SESSION_ID: str 
+    SESSION_MODE: str
+    SESSION_DRX_BEAM: str
+    CONFIG_FILE: str
+    CAL_DIR: str
+    STATUS: str  # e.g., "scheduled" "completed"
+
+
+class Settings(BaseModel):
+    time_loaded: str
+    user: str
+    filename: str
+    time_file: str
+
+
+class Calibrations(BaseModel):
+    time_loaded: str
+    filename: str
+    beam: str
+
+
+class Product(BaseModel):
+    time_loaded: str
+    filename: str
+    beam: str
+
+
+def create_db():
+    """Create database if it doesn't exist."""
+    conn = sqlite3.connect(DBPATH)
+    c = conn.cursor()
+    c.execute('''
+        CREATE TABLE IF NOT EXISTS sessions
+        (PI_ID text, PI_NAME text, PROJECT_ID text, SESSION_ID text, SESSION_MODE text, SESSION_DRX_BEAM text, CONFIG_FILE text, CAL_DIR text, STATUS text)
+    ''')
+    c.execute('''
+        CREATE TABLE IF NOT EXISTS settings
+        (time_loaded text, user text, filename text, time_file text)
+    ''')
+    c.execute('''
+        CREATE TABLE IF NOT EXISTS calibrations
+        (time_loaded text, filename text, beam text)
+    ''')
+    conn.commit()
+    conn.close()
+
+
+def read_sessions():
+    """Read all sessions from the database"""
+    conn = sqlite3.connect(DBPATH)
+    c = conn.cursor()
+    c.execute("SELECT * FROM sessions")
+    rows = c.fetchall()
+    conn.close()
+    return rows
+
+
+def read_settings():
+    """Read all settings from the database"""
+    conn = sqlite3.connect(DBPATH)
+    c = conn.cursor()
+    c.execute("SELECT * FROM settings")
+    rows = c.fetchall()
+    conn.close()
+    return rows
+
+
+def read_calibrations():
+    """Read all calibrations from the database"""
+    conn = sqlite3.connect(DBPATH)
+    c = conn.cursor()
+    c.execute("SELECT * FROM calibrations")
+    rows = c.fetchall()
+    conn.close()
+    return rows
+
+
+def add_session(sdffile: str):
+    """Parse SDF to create and add new session to the database."""
+    assert os.path.exists(sdffile), f"{sdffile} does not exist"
+    dd = parsesdf.sdf_to_dict(sdffile)
+    # convert lists to comma-separated strings
+    for key, value in dd['SESSION'].items():
+        if isinstance(value, list):
+            dd['SESSION'][key] = ', '.join(map(str, value))
+    session = Session(**dd['SESSION'], STATUS='scheduled')
+    conn = sqlite3.connect(DBPATH)
+    c = conn.cursor()
+    c.execute("INSERT INTO sessions VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)",
+              (session.PI_ID, session.PI_NAME, session.PROJECT_ID, session.SESSION_ID, session.SESSION_MODE, 
+               session.SESSION_DRX_BEAM, session.CONFIG_FILE, session.CAL_DIR, session.STATUS))
+    conn.commit()
+    conn.close()
+
+
+def add_settings(filename: str):
+    """Add settings to the database."""
+    assert os.path.exists(filename), f"{filename} does not exist"
+    user = getpass.getuser()
+    t_now = time.asctime(time.gmtime(time.time()))
```

Comment:
ideally, you want the caller to give you these things and make sure that the time entry is the time that the setting got loaded, instead of when this function gets called?